.circleci/config/@config.yml

```diff
@@ -24,6 +24,12 @@ commands:
             - ui/node_modules
   restore_go_cache:
     steps:
+      - run:
+          name: Allow circleci user to restore Go modules cache
+          command: |
+            set -eux -o pipefail
+            sudo mkdir /go
```

Comment:
```suggestion
            sudo mkdir -p /go
```

Reply:
also the go executor doesn't have sudo (just the go-machine one) so might need to switch the go-mod-download to a machine executor

Reply:
Thanks for the suggestion @nathanhruby! Good catch on the sudo as well, I'll think about how to approach this as would rather avoid using the machine executors except where really necessary.

Reply:
`which sudo ; if [ $? -eq 0 ] ; then SUDO=sudo ; else SUDO="" ; fi` should work

Reply:
Went for a slight variation on that, as I think `which sudo` would have caused failure on docker, as we use `set -e` amongst other options. See https://github.com/hashicorp/vault/pull/6894/files#diff-f3896cfad122301481e66fb84d437980R31

Reply:
Nope, still issues and I need to finish my day. I'll fix this tomorrow.

```
++ command -v sudo
+ SUDO=
+ mkdir -p /go
+ chown -R circleci:circleci /go
chown: invalid user: 'circleci:circleci'
Exited with code 1
```